src/wp-includes/blocks/index.php

```diff
@@ -39,6 +39,9 @@ function register_core_block_style_handles() {
 		'style'       => 'style',
 		'editorStyle' => 'editor',
 	);
+	if( current_theme_supports( 'wp-block-styles' ) ){
+		$style_fields['themeStyle'] = 'theme';
```

Comment:
Is the idea of a `themeStyle` field new? That is the part I'm a bit wary about to introduce this late in the cycle. It has more implications than a simple performance fix. This would be great to get feedback on by the editor team.